src/ServiceStack.Text/Support/StringSegmentExtensions.cs

```diff
@@ -119,62 +120,119 @@ enum ParseState
             TrailingWhite
         }
 
+        private static Exception GetOverflowException(Type type) => new OverflowException(String.Format(OverflowMessage, type.Name));
+
         public static sbyte ParseSByte(this StringSegment value)
         {
-            return JsConfig.UseSystemParseMethods
-                ? sbyte.Parse(value.Value, CultureInfo.InvariantCulture)
-                : (sbyte) ParseSignedInteger(value, sbyte.MaxValue, sbyte.MinValue);
+            try
+            {
+                return JsConfig.UseSystemParseMethods
+                    ? sbyte.Parse(value.Value, CultureInfo.InvariantCulture)
+                    : (sbyte) ParseSignedInteger(value, sbyte.MaxValue, sbyte.MinValue);
+            }
+            catch (OverflowException)
```

Comment:
These try/catches are needed only for compatibility to add to the message typename, which produces the error ("Int32", "Int64" etc). We can pass the type name to underlying method `ParseSignedInteger` and make the message from it, but in this case we have to add additional param in signature which would not used in mostly all cases of parse data, but requires to allocate it.